vocabularies/Common.xml

```diff
@@ -47,48 +48,48 @@
       </Term>
 
       <!-- Section: Versioning -->
-      <Term Name="ServiceVersion" Type="Edm.Int32" AppliesTo="Schema">
+      <Term Name="ServiceVersion" Type="Edm.Int32" Nullable="false" AppliesTo="Schema">
         <Annotation Term="Core.Description" String="1 for first version of a service, incremented when schema changes incompatibly and service is published with a different URI" />
       </Term>
-      <Term Name="ServiceSchemaVersion" Type="Edm.Int32" AppliesTo="Schema">
+      <Term Name="ServiceSchemaVersion" Type="Edm.Int32" Nullable="false" AppliesTo="Schema">
         <Annotation Term="Core.Description" String="0 for first schema version within a service version, incremented when schema changes compatibly" />
       </Term>
 
       <!-- Section: General Semantics -->
-      <Term Name="Label" Type="Edm.String">
+      <Term Name="Label" Type="Edm.String" Nullable="false">
         <Annotation Term="Core.Description" String="A short, human-readable text suitable for labels and captions in UIs" />
         <Annotation Term="Core.IsLanguageDependent" />
       </Term>
 
-      <Term Name="Heading" Type="Edm.String">
+      <Term Name="Heading" Type="Edm.String" Nullable="false">
         <Annotation Term="Core.Description" String="A short, human-readable text suitable for column headings in UIs" />
         <Annotation Term="Core.IsLanguageDependent" />
       </Term>
 
-      <Term Name="QuickInfo" Type="Edm.String">
+      <Term Name="QuickInfo" Type="Edm.String" Nullable="false">
         <Annotation Term="Core.Description" String="A short, human-readable text suitable for tool tips in UIs" />
         <Annotation Term="Core.IsLanguageDependent" />
       </Term>
 
-      <Term Name="Text" Type="Edm.String" AppliesTo="Property">
+      <Term Name="Text" Type="Edm.String" Nullable="true" AppliesTo="Property">
         <Annotation Term="Core.Description" String="A descriptive text for values of the annotated property. Value MUST be a dynamic expression when used as metadata annotation." />
         <Annotation Term="Core.IsLanguageDependent" />
       </Term>
 
-      <Term Name="TextFor" Type="Edm.PropertyPath" AppliesTo="Property">
+      <Term Name="TextFor" Type="Edm.PropertyPath" Nullable="false" AppliesTo="Property">
         <Annotation Term="Common.Experimental" />
         <Annotation Term="Core.RequiresType" String="Edm.String" />
         <Annotation Term="Core.Description" String="The annotated property contains a descriptive text for values of the referenced property." />
       </Term>
 
-      <Term Name="ExternalID" Type="Edm.String" AppliesTo="Property">
+      <Term Name="ExternalID" Type="Edm.String" Nullable="false" AppliesTo="Property">
```

Comment:
```suggestion
      <Term Name="ExternalID" Type="Edm.String" Nullable="true" AppliesTo="Property">
```
This should behave like `Text`, i.e., be nullable.

Reply:
Nullable IDs are a bit strange: what does it mean if two different instances both have `null` as their external ID?

Reply:
Not every instance must have an external ID, just like not every SAP unit of measurement has a UNCEFACT representation.

Reply:
An external ID is something that is for the end user an id of an object instance - this should be something, that the user can read and that's unique. Everything else seems to be not an ID to me.

Reply:
Good point: what would be visualized if the external ID is `null`, and how would that be visually distinguished from a non-null external ID?

Reply:
@bernhardthimmel: you "invented" the external ID: can/should it be nullable?

Reply:
I know it is already resolved but just for the record, yes I also think it should be nullable. E.g. there might also be issues with permissions so that you only get a technical id but that you are not able to see the text or the external id of the related object. Some colleagues in project management mentioned such a use case.
As for visualization, good question.